dag-store.js

```diff
@@ -4,13 +4,10 @@ import uuid from 'node-uuid';
 let nodes = (state = [], action = {}) => {
   switch(action.type) {
```

Comment:
The way I am using this you will never now at this component level what is being added to the state.  I am using this component in an application outside of this, for example if somebody adds a node that is going to run a hadoop job, I need the properties on that node about the hadoop job.  This has to do with the specific node state, not the state of the whole dag.

I will just keep using this on the fork then if you do not agree.  To me I think this is useful, if you want to protect certain attributes that will go on the nodes it's quite simple to do as I did here with your current required ones.  I am not sure why you aren't comfortable with the change and for me to implement this outside of the node keeping track of each node state is a lot more code and in my opinion more confusing.  I would think a lot of people using this library would want to add custom attributes to the nodes for similar reasons.